### PR TITLE
Support ODataLib/EdmLib/Spatial on ASP.NET 5

### PR DESCRIPTION
**Pull request by lewischeng-ms**
Linked issue #97

### CODE REVIEW

**Comment by PinpointTownes** [comment]
Wooo, nice :smile:

**Comment by lewischeng-ms** [comment]
@PinpointTownes Thanks for your attention. We have now produced signed binaries which work well so far. If you are interested, you can download them and help to test. The nuget source is: https://www.myget.org/feed/Packages/odlpreview Thanks again!

File: src/Microsoft.OData.Client/DataServiceContext.cs [hunk 1]
```diff
@@ -3479,7 +3466,7 @@ private static class ClientEdmModelCache
             static ClientEdmModelCache()
             {
                 IEnumerable<ODataProtocolVersion> protocolVersions =
-#if ASTORIA_LIGHT || PORTABLELIB // Silverlight does not support Enum.GetValues()
+#if PORTABLELIB
```
Comment (karataliu) [inline src/Microsoft.OData.Client/DataServiceContext.cs]: keep original comment.

File: src/Microsoft.OData.Client/HttpStack.cs [hunk 2]
```diff
@@ -8,27 +8,12 @@ namespace Microsoft.OData.Client
 {
     /// <summary>Represents the type of HTTP implementation to use when accessing the data service.Supported only by the WCF Data Services 5.0 client for Silverlight.</summary>
     [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1711:IdentifiersShouldNotHaveIncorrectSuffix", Justification = "Accurately describes what the enum represents, and also this class has already shipped as a public type.")]
-#if !ASTORIA_LIGHT
     internal enum HttpStack
```
Comment (karataliu) [inline src/Microsoft.OData.Client/HttpStack.cs]: This enum now has single value, seems we do not need it?

File: src/Microsoft.OData.Client/DataServiceContext.cs (hunk 1 above)
Comment (lewischeng-ms) [inline src/Microsoft.OData.Client/DataServiceContext.cs]: ok, will write a more meaningful comment.

File: src/Microsoft.OData.Client/HttpStack.cs (hunk 2 above)
Comment (lewischeng-ms) [inline src/Microsoft.OData.Client/HttpStack.cs]: remove this enum will bring a tons of code changes. we'd better do it in another task.

**Pull request merged**